codegen/src/generator/ec2.rs

```diff
@@ -408,7 +410,7 @@ fn generate_struct_field_serializers(shape: &Shape) -> String {
                 ",
                 field_name = generate_field_name(member_name),
                 member_shape_name = member.shape,
-                tag_name = member_name,
+                tag_name = tag_name,
```

Comment:
Yes, the previous code would use location_name but only for optional fields. After the changes the location_name is used regardless of whether a field is optional or not.